odl/solvers/functional/default_functionals.py

```diff
@@ -0,0 +1,731 @@
+# Copyright 2014-2016 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Default functionals defined on any (reasonable) space."""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+from builtins import super
+
+import numpy as np
+
+from odl.solvers.functional.functional import Functional
+from odl.operator.operator import Operator
+from odl.solvers.advanced.proximal_operators import (
+    proximal_l1, proximal_cconj_l1, proximal_l2, proximal_cconj_l2,
+    proximal_l2_squared)
+
+from odl.operator.default_ops import (ZeroOperator, IdentityOperator)
+
+
+__all__ = ('L1Norm', 'L2Norm', 'L2NormSquare', 'ZeroFunctional',
+           'ConstantFunctional')
+
+
+# TODO: Implement some of the missing gradients
+
+class L1Norm(Functional):
+
+    """The functional corresponding to L1-norm.
+
+    The L1-norm, ``||x||_1``,  is defined as the integral/sum of ``|x|``.
+
+    References
+    ----------
+    Wikipedia article on `Norms
+        <https://en.wikipedia.org/wiki/Norm_(mathematics)>`_.
+    """
+
+    def __init__(self, domain):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        domain : `LinearSpace`
```

Comment:
The l1 norm on product spaces i.m.o. deserves its own class, or at  least quite some attention. We have the exact same problem in the current l1 proximal, where we solve it in  rather clunky way. ProxImaL solves it by adding the concept of  a ["group l1 norm"](https://github.com/comp-imaging/ProxImaL/blob/master/proximal/prox_fns/group_norm1.py) which does this is a (i.m.o.) nicer and more elegant way.

Regarding your example the result is debatable, i.m.o. it should either by `sqrt(2)` or `1` since it is `|| (||1||_q, ||1||_q) ||_p` where the naming implies that at least one of `p` and `q` is 1.

Options are then

```
|| (||1||_2, ||1||_2) ||_1 = || (1, 1) ||_1 = 2
|| (||1||_1, ||1||_1) ||_1 = || (1, 1) ||_1 = 2
|| (||1||_1, ||1||_1) ||_2 = || (1, 1) ||_2 = sqrt(2)
```

where the first two cases would have had different results if the spaces were not the unit interval.

Anyway this ambiguity is itself a strong argument for perhaps separating the powerspace situation from the others. If we allow this in powerspaces I recommend going for the `p=q=1` approach.